WalletWasabi.Standard/CoinJoin/Coordinator/Coordinator.cs

```diff
@@ -216,38 +218,50 @@ public async Task ProcessConfirmedTransactionAsync(Transaction tx)
 			}
 		}
 
-		public async Task MakeSureTwoRunningRoundsAsync(Money feePerInputs = null, Money feePerOutputs = null)
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+		public async Task MakeSureInputregistrableRoundRunningAsync()
 		{
 			using (await RoundsListLock.LockAsync().ConfigureAwait(false))
 			{
-				int runningRoundCount = Rounds.Count(x => x.Status == CoordinatorRoundStatus.Running);
-
-				int confirmationTarget = await AdjustConfirmationTargetAsync(lockCoinJoins: true).ConfigureAwait(false);
-
-				if (runningRoundCount == 0)
+				if (!Rounds.Any(x => x.Status == CoordinatorRoundStatus.Running && x.Phase == RoundPhase.InputRegistration))
 				{
-					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					int confirmationTarget = await AdjustConfirmationTargetAsync(lockCoinJoins: true).ConfigureAwait(false);
+					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate, TimeSpan.FromSeconds(RoundConfig.InputRegistrationTimeout));
 					round.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
 					round.StatusChanged += Round_StatusChangedAsync;
-					await round.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
-					Rounds.Add(round);
-
-					var round2 = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
-					round2.StatusChanged += Round_StatusChangedAsync;
-					round2.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
-					await round2.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
-					Rounds.Add(round2);
-				}
-				else if (runningRoundCount == 1)
-				{
-					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
-					round.StatusChanged += Round_StatusChangedAsync;
-					round.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
-					await round.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
+					round.PhaseChanged += Round_PhaseChangedAsync;
+					await round.ExecuteNextPhaseAsync(RoundPhase.InputRegistration).ConfigureAwait(false);
 					Rounds.Add(round);
+					SchedulePushAsync(round.InputRegistrationTimeout);
```

Comment:
Instead of the `#pragma warning disable` you can do 
```suggestion
					_ = SchedulePushAsync(round.InputRegistrationTimeout);
```